sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/CommonUtil.kt

```diff
@@ -0,0 +1,246 @@
+                googleApi.getErrorDialog(context, result, PLAY_SERVICES_RESOLUTION_REQUEST)?.show()
+            }
+            return false
+        }
+        return true
+    }
+
+    /**
+     * Get the UTC date format from the simple date format, i.e. "yyyy-MM-dd'T'HH:mm:ssX"
+     *
+     * @return the date format
+     */
+    @SuppressLint("SimpleDateFormat")
+    fun getUTCDateFormat(): DateFormat {
+        val dateFormat: DateFormat = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX")
+        } else {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss")
+        }
+        val timeZone = TimeZone.getTimeZone("UTC")
+        dateFormat.timeZone = timeZone
+        return dateFormat
+    }
+
+    /**
+     * Get the UTC date from a sample date string
+     *
+     * @param dateString the date string.
+     * @return the date object based on the date string
+     */
+    @SuppressLint("SimpleDateFormat")
+    fun getUTCDate(dateString: String?): Date? {
+        val dateFormat: DateFormat = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX")
+        } else {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss")
+        }
+        val timeZone = TimeZone.getTimeZone("UTC")
+        dateFormat.timeZone = timeZone
+        return dateFormat.parse(dateString ?: "2021-12-01T00:00:00Z")
+    }
+
+    /**
+     * Get the resource id from drawable folder.
+     *
+     * @param context the application context.
+     * @param resourceName the resource name
+     * @return the drawable resource id
+     */
+    fun getDrawableResourceId(context: Context, resourceName: String?): Int {
+        return context.resources.getIdentifier(resourceName, "drawable", context.packageName)
+    }
+
+    /**
+     * Get the resource id from raw folder.
+     *
+     * @param context the application context.
+     * @param resourceName the resource name
+     * @return the raw resource id
+     */
+    fun getRawResourceId(context: Context, resourceName: String?): Int {
+        return context.resources.getIdentifier(resourceName, "raw", context.packageName)
+    }
+
+    /**
+     * Parse the color string, and return the corresponding color-int.
+     *
+     * @param colorString the color string.
+     * @return the color int value
+     */
+    @ColorInt
+    fun getColorValue(colorString: String?): Int {
+        return try {
+            Color.parseColor(colorString)
+        } catch (exception: Exception) {
+            logger.error(exception, "Failed to parse color", exception)
+            -1
+        }
+    }
+
+    /**
+     * Get the desired file path exists in assets folder.
+     *
+     * @param context the application context.
+     * @param fileName the file name
+     * @return the asset path of the file
+     */
+    fun getAssetsFilePath(context: Context, fileName: String?): String? {
```

Comment:
Can we make this as extension of `Context` class?

Reply:
updated